### PR TITLE
use a new library to parse command line args

### PR DESCRIPTION
**Pull request by notrodes**
also some minor cleaning

### CODE REVIEW

File: serve.js [hunk 1]
```diff
@@ -16,26 +17,18 @@ const args = require('minimist')(process.argv.slice(2));
...
+    .parse();
 
 // ------------ Web Server -------------------
-const port = 8080;
-
 const app = express();
-app.use(compression({ filter: (..._) => true }));
-app.listen(port, () => console.log(`Aspine listening on port ${port}!`));
+app.use(compression());
+app.listen(program.port, () => console.log(`Aspine listening on port ${program.port}!`));
```
Comment (psvenk) [inline serve.js]: @notrodes Why did you remove the custom `filter` function? I had included it because `compression` by default does not compress all traffic and so feeding it a function that always returns `true` causes all traffic to be compressed.

File: serve.js [hunk 2]
```diff
@@ -47,22 +40,19 @@ if(!(args.hasOwnProperty('insecure') || args._.includes("insecure"))) {
...
             // OK, continue
             return next();
         }
         // handle port numbers if you need non defaults
-        // res.redirect('https://' + req.host + req.url); // express 3.x
         res.redirect('https://' + req.hostname + req.url); // express 4.x
-    }
+    }); // at top of routing calls
+
+    http.createServer(app).listen(8090);
```
Comment (psvenk) [inline serve.js]:
```suggestion
```
I think this line can be removed because we do not serve HTTP on production.

File: serve.js [hunk 3]
```diff
@@ -47,22 +40,19 @@ if(!(args.hasOwnProperty('insecure') || args._.includes("insecure"))) {
...
+    https.createServer(credentials, app).listen(4430, () =>//443
+        console.log('HTTPS Server running on port 4430') //443
+    );
```
Comment (psvenk) [inline serve.js]: Remove outdated comments
```suggestion
    https.createServer(credentials, app).listen(4430, () =>
        console.log('HTTPS Server running on port 4430')
    );
```

File: serve.js [hunk 4]
```diff
@@ -47,22 +40,19 @@ if(!(args.hasOwnProperty('insecure') || args._.includes("insecure"))) {
...
-    function ensureSecure(req, res, next){
-        if(req.secure){
+    app.all('*', (req, res, next) => {
+        if (req.secure) {
             // OK, continue
             return next();
         }
         // handle port numbers if you need non defaults
-        // res.redirect('https://' + req.host + req.url); // express 3.x
         res.redirect('https://' + req.hostname + req.url); // express 4.x
```
Comment (psvenk) [inline serve.js]: Remove outdated/unnecessary comment
```suggestion
        res.redirect('https://' + req.hostname + req.url);
```

**Review by psvenk — CHANGES_REQUESTED** [review]
Additional changes which I cannot put as inline comments (because GitHub won't let me comment on lines of code that were not changed):
- `npm install commander` and commit `package.json` and `package-lock.json`.
- `minimist` is no longer needed, so the `require('minimist')` can be removed and we can also `npm uninstall minimist`.
- The variables `privateKey`, `certificate`, and `ca` are not needed and can be folded into the object literal for `credentials`.
- The version number should come dynamically from `package.json`.

File: serve.js [hunk 5]
```diff
@@ -16,26 +17,18 @@ const args = require('minimist')(process.argv.slice(2));
...
-const port = 8080;
-
 const app = express();
-app.use(compression({ filter: (..._) => true }));
-app.listen(port, () => console.log(`Aspine listening on port ${port}!`));
+app.use(compression());
+app.listen(program.port, () => console.log(`Aspine listening on port ${program.port}!`));
 
-if(!(args.hasOwnProperty('insecure') || args._.includes("insecure"))) {
+if(!program.insecure) {
```
Comment (psvenk) [inline serve.js]: Fix spacing
```suggestion
if (!program.insecure) {
```

Comment (psvenk) [inline serve.js]: This breaks the usage of `npm start insecure`, meaning that we need to update documentation to use `npm start -- -i` instead (which IMO is not very elegant).

Comment (notrodes) [reply serve.js]: no, -i or --insecure, which is normal

File: serve.js (hunk 1 above)
Comment (psvenk) [reply serve.js]: By the way, I think it should be fine to change the function from `(..._) => true` to the more concise `() => true`, but we should double-check that everything is being compressed.

File: serve.js (hunk 5 above)
Comment (psvenk) [reply serve.js]: My main concern is not inelegance but that documentation and start scripts need to be updated to use `npm start -- -i`.

File: serve.js (hunk 1 above)
Comment (notrodes) [reply serve.js]: i checked and the default compresses based on what data type it is being used. I though that would be better than everything being compressed.

Comment (psvenk) [reply serve.js]: I guess the default algorithm changed since I did this in #125, because now the difference in amount of data transferred is negligible between full compression and default compression. In that case, `compression()` is indeed probably better.

**Review by psvenk — APPROVED** [review]
